### PR TITLE
Apply cargo clippy and cargo fmt

### PR DESCRIPTION
**Pull request by Byron**
What's missing is running these on CI, which I refrained from doing in this PR to not conflate things.

### CODE REVIEW

**Comment by Byron** [comment]
It looks like [CI fails](https://travis-ci.org/github/Marwes/combine/jobs/734557476#L520) as `cargo doc` is run with rust 1.40. Recently clippy started to suggest the [matches!()](https://caniuse.rs/features/matches_macro) macro, which is available in 1.42 and above.

Would you like me to remove usages of the `matches!()` macro, or rather allow docs to be generated with rustc 1.42?

**Comment by Marwes** [comment]
Might be easiest to remove the use of matches!, rather than adjusting ci to skip doctests on 1.40. Either way works for me though!

**Comment by Byron** [comment]
Alright! I think if travis runs the `.travis.yaml` from the PR, it should work next time with upping the minimal version used to 1.42 . I really like `matches!()` and am glad clippy now informs about it - wasn't aware of it's existence at all till now.

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ rust:
 - nightly
 - beta
 - stable
-- 1.40.0
+- 1.42.0
```
Comment (Marwes) [inline .travis.yml:16]: 1.40 is explicitly the minimum supported rust version for the current combine major version. I don't necessarily want to keep this minimum version until the next breaking (major version) but I do try to stick to it unless actually necessary.

For something like documentation though I would rather keep the minimum version as is.

So either skip using matches! or disable doc tests on 1.40 would be acceptable to me.

Comment (Byron) [reply .travis.yml]: Alright, the next commit attempts to disable doc-tests for 1.40, and should that fail or run into other issues, I will remove `matches!()` all together.

**Comment by Marwes** [comment]
Thanks!

**Pull request merged**